lib/ansible/plugins/inventory/openstack.py

```diff
@@ -192,9 +210,14 @@ def parse(self, inventory, loader, path, cache=True):
 
             expand_hostvars = self._config_data.get('expand_hostvars', False)
             fail_on_errors = self._config_data.get('fail_on_errors', False)
+            filters = self.templar.template(self._config_data.get('filters', None))
```

Comment:
I think instead of templating this particular option here it should be added in a more general way. Have you seen #58288?

Reply:
Hi!

I haven't seen that PR but I'll agree that is the right way to do this.

This new option doesn't really make sense if templating is not included, In that case some other templating mechanism / scripting must be used to achieve dynamic / environment aware filtering.

How about merging this as it is for now and removing templating after #58288 is merged? There seems to failing tests and I wouldn't want to wait if it makes to 2.9.

Reply:
:+1: this seems like a sensible approach.

Reply:
I marked this as resolved based on @cloudnull's comment and because of https://github.com/ansible/ansible/pull/58288 is still failing in CI.

Reply:
I marked as unresolved because #60715 based on #58288, and it is not failing CI. It just needs reviews. It also allows you to use inline vaults in the inventory config.